ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -295,6 +300,25 @@ private Response executeStatements(
     }
   }
 
+  private void waitForPreviousDistributedStatementToBeHandled(final KsqlEntityList entities) {
+    final ArrayList<KsqlEntity> reversed = new ArrayList<>(entities);
+    Collections.reverse(reversed);
+
+    reversed.stream()
+        .filter(e -> e instanceof CommandStatusEntity)
+        .map(cs -> ((CommandStatusEntity) cs).getCommandSequenceNumber())
```

Comment:
```
.map(CommandStatusEntity.class::cast)
.map(CommandStatusEntity::getSequenceNumber)
```
The jury is still out on whether or not I prefer that or what you currently have, just wanted to throw it out there :)

Reply:
I'm definitely in your camp. Think this was a refactoring artefact, not by design.

Reply:
Change will be in next PR.